libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1392,15 +1393,15 @@ void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const&
 				checkExpressionAssignment(*types[i], *tupleExpression->components()[i]);
 			}
 	}
-	else if (_type.category() == Type::Category::Mapping)
+	else if (_type.nameable() && _type.containsNestedMapping())
```

Comment:
`nameable()` was added here to fix a certain test case from calling `nestedMapping()` for `RationalNumberType`.  The expression causing the problem: `(((((((((((,2),)),)),),))=4)));`

Test case: `syntaxTests/tupleAssignments/tuple_in_tuple_long.sol`

Reply:
What happens if you replace the `2` by a variable of mapping type?

Reply:
Here are the errors produced:
```
Error: Types in storage containing (nested) mappings cannot be assigned to.
 --> <stdin>:5:17:
  |
5 |     (((((((((((,m),)),)),),))=4)));
  |                 ^

Error: Type int_const 4 is not implicitly convertible to expected type tuple(tuple(tuple(tuple(tuple(,mapping(uint256 => uint256)),),),),).
 --> <stdin>:5:31:
  |
5 |     (((((((((((,m),)),)),),))=4)));
  |                               ^

Error: Tuple component cannot be empty.
 --> <stdin>:5:8:
  |
5 |     (((((((((((,m),)),)),),))=4)));
  |        ^^^^^^^^^^^^^^^^^^^^^^^^

Error: Tuple component cannot be empty.
 --> <stdin>:5:7:
  |
5 |     (((((((((((,m),)),)),),))=4)));
  |       ^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Tuple component cannot be empty.
 --> <stdin>:5:6:
  |
5 |     (((((((((((,m),)),)),),))=4)));
  |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```